File: Lab 1.1/Lab11.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

vector<string> createText();

void createFile(string,vector<string>);

void outputFile(string);

void createSecondFile(vector<string>,string);

char findCharecter(string);

int main(){
    string Name1="File1.txt", Name2="File2.txt";
    vector<string> Text;

    Text=createText();
    createFile(Name1,Text);
    outputFile(Name1);

    createSecondFile(Text,Name2);
    outputFile(Name2);

    system("pause");
}

vector<string> createText(){
    cout<<"While input the text press Enter \
    \nto input new line and \
    \nCtrl+D to finish input "<<endl;
    cout<<"\nEnter texr:"<<endl;
    string str;
    int code=4;
    vector<string> text;
    while(int(str[0])!=code){
        getline(cin,str);
        text.push_back(str);
    }
    return text;
}

void createFile(string name,vector<string> text){
    ofstream outFile(name, ios::trunc);

    if(!outFile.is_open()) cout<<"Cannot create the File!"<<endl;

    for (int i = 0; i < text.size()-1; i++)
    {
        outFile<<text[i]<<endl;
    }
    
    outFile.close();
}

void outputFile(string name){
    cout<<"\nFile \""<<name<<"\":"<<endl;

    ifstream inFile(name);
    if(!inFile.is_open()) cout<<"Cannot open the File!"<<endl;
    string str;

    while(!inFile.eof()){
        getline(inFile,str);
        cout<<str<<endl;
    }

    inFile.close();
}

void createSecondFile(vector<string> text, string name2){
    ofstream outFile(name2, ios::trunc);

    string str;

    if(!outFile.is_open()) cout<<"Cannot create the file!"<<endl;
    for(int i=0;i<text.size()-1;i++){
        str=text[i];
        str = str + ' ' + findCharecter(str);
        outFile<<str<<endl;
    }

    outFile.close();
}

char findCharecter(string str){
    int pos=0;
    for (int i = 0; str[i]!=' '; i++)
    {
        pos++;
    }
    return str[--pos];
}

File: Lab 1.1/Lab11.py
def import_text():
    print('While input the text press Enter \
    \nto input new line and \
    \nCtrl+D to finish input \
    \n\nEnter text:')
    import sys
    text=sys.stdin.read()
    text[:len(text)-1]
    return text

def create_file(text,name):
    with open(name,'wt') as fwrite:
        fwrite.write(text)

def output_file(name):
    with open(name,'rt') as fread:
        text=fread.read()
        print("\nFile \"",name,"\":\n",text,sep="")

def create_second_file(name1,name2):
    fread=open(name1,'rt')
    fwrite=open(name2,'wt')

    while True:
        line=fread.readline()
        if not line:
            break
        line=line.strip('\n') + ' ' + str(find_charecter(line))
        fwrite.write(line+'\n')
    
    fread.close()
    fwrite.close()

def find_charecter(line):
    for i in range(len(line)):
        if line[i]==' ' or line[i]=='.' or line[i]==',' or line[i]=='!' or line[i]=='?' or line[i]==':' or line[i]==';':
            return line[i-1]



Name1="Text1.txt"
Name2="Text2.txt"

Text=import_text()
create_file(Text,Name1)
output_file(Name1)

create_second_file(Name1,Name2)
output_file(Name2)

input("Press Enter to continue...")
